configman/config_manager.py

```diff
@@ -675,7 +686,9 @@ def _overlay_expand(self):
                 if key in all_reference_values:
                     # make sure that this value gets propagated to keys
                     # even if the keys have already been overlaid
-                    known_keys -= set(all_reference_values[key])
+                    finished_keys -= set(
+                        all_reference_values[key]
+                    )
```

Comment:
I don't think that works.  This code fragment is working with a specific key and testing it is a reference value, and if so, mark the keys that depend on it as unfinished.  Your code fragment is looping over all keys that are reference values and marking them all as unfinished.  That would likely cause an infinite loop.